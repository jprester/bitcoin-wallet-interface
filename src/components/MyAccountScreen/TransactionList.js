import React from 'react';
import { uniqueId } from 'lodash';
import PropTypes from 'prop-types';

import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
import ListItemText from '@material-ui/core/ListItemText';

const TransactionList = ({ transactionList, classes }) => {
  if (!transactionList.length) {
    return (
      <p> List is empty </p>
    );
  }

  return (transactionList.length &&
  <List>
    {transactionList.map(transaction => (
      <ListItem key={uniqueId('list-item')} dense disableGutters divider className={classes.listItem}>
        <ListItemText className={classes.transactionAdress} primary={transaction.address} />
        <ListItemSecondaryAction className={classes.transactionAmount}>
          {transaction.amount} BTC
        </ListItemSecondaryAction>
      </ListItem>
    ))}
  </List>
  );
};

export default TransactionList;


TransactionList.propTypes = {
  transactionList: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  classes: PropTypes.shape({}).isRequired,
};

