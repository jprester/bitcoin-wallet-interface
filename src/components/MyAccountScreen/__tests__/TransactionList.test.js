import React from 'react';
import { shallow } from 'enzyme';
import TransactionList from '../TransactionList';

const wrapper =
  shallow((<TransactionList classes={{ listItem: {} }} transactionList={[{ address: '1Cs4wu6pu5qCZ35bSLNVzGyEx5N6uzb', amount: '1.2' }]} />));

describe('TransactionList', () => {
  it('should render TransactionList correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
