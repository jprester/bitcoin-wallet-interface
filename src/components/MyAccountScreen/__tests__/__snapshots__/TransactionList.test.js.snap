// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionList should render TransactionList correctly 1`] = `
<WithStyles(List)>
  <WithStyles(ListItem)
    className={Object {}}
    dense={true}
    disableGutters={true}
    divider={true}
    key="list-item1"
  >
    <WithStyles(ListItemText)
      primary="1Cs4wu6pu5qCZ35bSLNVzGyEx5N6uzb"
    />
    <WithStyles(ListItemSecondaryAction)>
      1.2
       BTC
    </WithStyles(ListItemSecondaryAction)>
  </WithStyles(ListItem)>
</WithStyles(List)>
`;
